echo_server.py

```diff
@@ -0,0 +1,23 @@
+#! /usr/bin/env python
+"""Echo server in socket connection: receives and sends back a message."""
+import socket
+
+
+if __name__ == '__main__':
+    """Run from terminal, this will recieve a messages and send them back."""
+    server_socket = socket.socket(socket.AF_INET, socket.SOCK_STREAM,
+                                  socket.IPPROTO_IP)
+    server_socket.bind(('127.0.0.1', 50000))
+    server_socket.listen(1)
+    buffsize = 32
+    while True:
+        msg = ''
+        done = False
+        conn, addr = server_socket.accept()
+        while not done:
+            msg_part = conn.recv(buffsize)
+            msg += msg_part
+            if len(msg_part) < buffsize:
+                done = True
+                conn.sendall(msg)
+                conn.close()
```

Comment:
I'd like to see some code that allows you to close the server socket when you terminate this script.  Otherwise, very good.

Also, do you really need to accumulate the entire message before sending it back?  Or can you simply send back the buffer you received as soon as you receive it?